dev-tools/omdb/src/bin/omdb/db.rs

```diff
@@ -1426,6 +1528,431 @@ async fn cmd_db_snapshot_info(
+        println!("old region id: {}", request.old_region_id);
+        println!("new region id: {:?}", request.new_region_id);
+        println!();
+
+        if let Some(new_region_id) = request.new_region_id {
+            println!();
+
+            // Find the most recent upstairs repair notification where the
+            // downstairs being repaired is a "new" region id. This will give us
+            // the most recent repair id.
+            let maybe_repair: Option<UpstairsRepairNotification> = datastore
+                .pool_connection_for_tests()
+                .await?
+                .transaction_async(|conn| async move {
+                    use db::schema::upstairs_repair_notification::dsl;
+
+                    dsl::upstairs_repair_notification
+                        .filter(dsl::region_id.eq(new_region_id))
+                        .filter(
+                            dsl::notification_type
+                                .eq(UpstairsRepairNotificationType::Started),
+                        )
+                        .order_by(dsl::time.desc())
+                        .limit(1)
+                        .first_async(&conn)
+                        .await
+                        .optional()
+                })
+                .await?;
+
+            if let Some(repair) = maybe_repair {
+                let maybe_repair_progress: Option<UpstairsRepairProgress> =
+                    datastore
+                        .pool_connection_for_tests()
+                        .await?
+                        .transaction_async(|conn| async move {
+                            use db::schema::upstairs_repair_progress::dsl;
+
+                            dsl::upstairs_repair_progress
+                                .filter(dsl::repair_id.eq(repair.repair_id))
+                                .order_by(dsl::time.desc())
+                                .limit(1)
+                                .first_async(&conn)
+                                .await
+                                .optional()
+                        })
+                        .await?;
+
+                if let Some(repair_progress) = maybe_repair_progress {
+                    let bar = ProgressBar::with_draw_target(
+                        Some(repair_progress.total_items as u64),
+                        ProgressDrawTarget::stdout(),
+                    )
+                    .with_style(ProgressStyle::with_template(
+                        "progress:\t{wide_bar:.green} [{pos:>7}/{len:>7}]",
+                    )?)
+                    .with_position(repair_progress.current_item as u64);
+
+                    bar.abandon();
+                }
+            }
+        }
+
+        println!();
+    }
+
+    Ok(())
+}
+
+/// Show details for a single region replacement
+async fn cmd_db_region_replacement_info(
+    _opctx: &OpContext,
+    datastore: &DataStore,
+    _fetch_opts: &DbFetchOptions,
```

Comment:
If we don't need these, we should probably get rid of them.

Reply:
addressed in a155af913